lib/front_desk.rb

```diff
@@ -0,0 +1,118 @@
+require_relative 'room'
+require_relative 'reservation'
+require_relative 'block'
+
+require 'date'
+require 'pry'
+
+class FrontDesk
+  attr_reader :reservations, :rooms, :blocks
+
+  def initialize
+    @rooms = load_rooms
+    @reservations = []
+    @blocks = []
+
+  end
+
+  def load_rooms
+    rooms = []
+    20.times do |i|
+      rooms << Room.new((i + 1))
+    end
+    return rooms
+  end
+
+  def list_all_rooms
+    rooms.each do |room|
+      room
+    end
+  end
+
+  def create_reservation(room_number, start_day, end_day, nightly_cost)
+    check_availability(room_number, start_day, end_day)
+    new_reservation = Reservation.new(room_number, start_day, end_day,nightly_cost)
+    reservations << new_reservation
+    room = find_room(room_number)
+    room.add_reservation_to_room(new_reservation)
+
+  end
+
+  def find_room(room_number)
+    return @rooms.find {|room| room.room_number == room_number}
+  end
+
+  def find_by_date(date)
```

Comment:
This method only finds reservations that **start** on the given date.  Instead you should return reservations are occur **during** that date.